bugsnag-plugin-android-ndk/src/main/jni/metadata.c

```diff
@@ -606,15 +287,16 @@ void bsg_populate_metadata_obj(JNIEnv *env, bugsnag_report *dst,
                                                section_key);
   char *name = (char *) bsg_safe_get_string_utf_chars(env, section_key);
   if (name != NULL) {
-    bsg_populate_metadata_value(env, dst, jni_cache, section, name, _value);
+    populate_metadata_value(env, dst, section, name, _value);
     bsg_safe_release_string_utf_chars(env, section_key, name);
   }
   bsg_safe_delete_local_ref(env, _value);
 }
 
-void bsg_populate_metadata_section(JNIEnv *env, bugsnag_report *report,
-                                   jobject metadata, bsg_jni_cache *jni_cache,
-                                   jobject keylist, int i) {
+static void populate_metadata_section(JNIEnv *env, bugsnag_report *report,
+                                      jobject metadata, jobject keylist, int i) {
+  type_bsg_jni_cache *jni_cache = bsg_get_jni_cache();
```

Comment:
IMO it's not particularly clear which methods have checked `bsg_is_jni_cache_valid`. I think it'd be safer to either always call `bsg_is_jni_cache_valid`, or to create a convention where `type_bsg_jni_cache` is passed as a parameter when the cache has already been checked.